lib/src/components/NxFilterDropdown/NxFilterDropdown.tsx

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { Ref, useRef } from 'react';
+import { propEq, any } from 'ramda';
+import classnames from 'classnames';
+
+import { Props, propTypes } from './types';
+import NxCheckbox from '../NxCheckbox/NxCheckbox';
+import MultiSelectCounter from '../Counter/MultiSelectCounter';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+import { faFilter, faTimes } from '@fortawesome/free-solid-svg-icons';
+import NxDropdown from '../NxDropdown/NxDropdown';
+import DataItem from '../../util/DataItem';
+import forwardRef from '../../util/genericForwardRef';
+
+import './NxFilterDropdown.scss';
+
+export { Props };
+
+function NxFilterDropdownRender<T extends string | number = string>(props: Props<T>, ref: Ref<HTMLDivElement>) {
+  const {
+        onChange,
+        selectedIds,
+        isOpen,
+        onToggleCollapse,
+        options,
+        className: classNameProp,
+        showReset,
+        placeholder,
+        ...attrs
+      } = props,
+      menuRef = useRef<HTMLDivElement>(null),
+      className = classnames('nx-filter-dropdown', classNameProp);
+
+  // Throw an error if one of the selectedIds is not part of the available options
+  selectedIds.forEach(itemId => {
+    if (!any(propEq('id', itemId), options)) {
+      throw new Error(`You are attempting to select "${itemId}", but it is not part of the available options`);
+    }
+  });
+
+  function toggle(id: T) {
+    const newSelected = new Set(selectedIds);
+    if (newSelected.has(id)) {
+      newSelected.delete(id);
+    }
+    else {
+      newSelected.add(id);
+    }
+    onChange(newSelected, id);
+  }
+
+  // Avoid closing dropdown when a click occurs within it (e.g. when a checkbox is toggled)
+  function onCloseClick(evt: MouseEvent) {
+    const { target } = evt,
+        menuEl = menuRef.current;
+
+    if (menuEl && target instanceof Element && menuEl.contains(target)) {
+      evt.preventDefault();
+    }
+  }
+
+  function onResetClick() {
+    onChange(new Set());
+  }
+
+  const renderOption = ({ id, displayName }: DataItem<T>) => (
+    <NxCheckbox key={id}
+                onChange={() => toggle(id)}
+                isChecked={selectedIds.has(id)}>
+      {displayName}
+    </NxCheckbox>
+  );
+
+  const renderCounter = () => <MultiSelectCounter options={options} selectedIds={selectedIds} />,
+      dropdownLabel = (
+        <>
+          <NxFontAwesomeIcon className="nx-filter-dropdown__filter-icon" icon={faFilter} />
+          { selectedIds.size ? renderCounter() : <span>{placeholder ?? 'Filter'}</span> }
+        </>
+      );
+
+  const resetBtn = (
```

Comment:
> It seems that the faTimes icon is not of size 16x16px as per the Sketch mockup.

Actually I need to check with Design about the use of that icon vs `NxCloseIcon`.  This spurred the discussion about the two but after the decision was made to keep the custom close icon we forgot to revisit whether that should be used here instead of `faTimes`.

Reply:
> Also, it looks like the spacing for the reset in the mockup is a bit different? The spacing between "Item 3" and "Reset" is 13px according to Sketch. Not sure if that's the case with the component?

I'll check that with Design as well.  Looking closely at the mockup, aside from the question about which close icon we should be using and how big it should be, the alignment between the checkboxes and their labels also seems to be off (in the mockup) making it impossible to tell what the distance between the checkbox and the reset button should be

Reply:
> I'll check that with Design as well. Looking closely at the mockup, aside from the question about which close icon we should be using and how big it should be, the alignment between the checkboxes and their labels also seems to be off (in the mockup) making it impossible to tell what the distance between the checkbox and the reset button should be

Updated after design tweaks in 84eaaaa6.  Make sure you check the latest mockups